rust/worker/src/index/hnsw.rs

```diff
@@ -647,4 +674,50 @@ pub mod test {
             i += 1;
         }
     }
+
+    #[test]
+    fn it_can_add_and_query_with_allowed_and_disallowed_ids() {
+        let n = 1000;
+        let d: usize = 960;
+        let distance_function = DistanceFunction::Euclidean;
+        let tmp_dir = tempdir().unwrap();
+        let persist_path = tmp_dir.path().to_str().unwrap().to_string();
+        let index = HnswIndex::init(
+            &IndexConfig {
+                dimensionality: d as i32,
+                distance_function: distance_function,
+            },
+            Some(&HnswIndexConfig {
+                max_elements: n,
+                m: 16,
+                ef_construction: 100,
+                ef_search: 100,
+                random_seed: 0,
+                persist_path: persist_path,
+            }),
+            Uuid::new_v4(),
+        );
+
+        let index = match index {
+            Err(e) => panic!("Error initializing index: {}", e),
+            Ok(index) => index,
+        };
+
+        let data: Vec<f32> = utils::generate_random_data(n, d);
+        let ids: Vec<usize> = (0..n).collect();
+
+        (0..n).into_iter().for_each(|i| {
+            let data = &data[i * d..(i + 1) * d];
+            index.add(ids[i], data);
+        });
+
+        // Query the data
+        let query = &data[0..d];
+        let allow_ids = &[0, 2, 3];
+        let disallow_ids = &[];
```

Comment:
can we add test for the disallow case and also with / without truncation.

Also - should we error if allow_ids and disallow_ids overlap?